### ISSUE TITLE
`next export` with `exportPathMap` in `next.config.js` creates a duplicate 404.html.html

### ISSUE THREAD

**Issue by mdubourg001** [vercel/next.js#36855]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
$ yarn next info

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:41 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.1.0
      npm: 8.8.0
      Yarn: 3.2.0
      pnpm: N/A
    Relevant packages:
      next: 12.1.6
      react: 18.1.0
      react-dom: 18.1.0
```

### Describe the Bug

This is actually a following of https://github.com/vercel/next.js/issues/33464.

Running `yarn build && yarn export` while using an updated `exportPathMap` in `next.config.js` leads to the creation of an extra `404.html.html` file in export. It only happens if the object returned from `exportPathMap` doesn't have a key named `/404`.

https://github.com/vercel/next.js/blob/e8a8220d49434021f2fb9975a6ec7081043b97b4/packages/next/export/index.ts#L421

For example, my goal was to create each page in an `index.html` in a directory with the same name (ie. `/about.html` becomes `/about/index.html`):

 When using this `next.config.js`, the export produces an extra `404.html.html` file (ie. the bug happens)...

```js
// next.config.js

module.exports = {
  exportPathMap: async function (defaultPathMap) {
    const pathMap = {};

    for (const [path, config] of Object.entries(defaultPathMap)) {
      if (path === "/") {
        pathMap[path] = config;
      } else {
        pathMap[`${path}/index`] = config;
      }
    }

    return pathMap;
  }
}
```

... while when using this one, it doesn't happen:

```js
// next.config.js

module.exports = {
  exportPathMap: async function (defaultPathMap) {
    // 🚩the only difference is here, we spread the default pathMap
    const pathMap = { ...defaultPathMap };

    for (const [path, config] of Object.entries(defaultPathMap)) {
      if (path === "/") {
        pathMap[path] = config;
      } else {
        pathMap[`${path}/index`] = config;
      }
    }

    return pathMap;
  }
}
```

### Expected Behavior

The `404.html.html` page should not be created when using a custom `exportPathMap` in `next.config.js`.

### To Reproduce

1. Open https://stackblitz.com/edit/nextjs-rxdeeq
2. Run `npx next build && npx next export` in the terminal
3. `404.html.html` has been created in the `out/` directory

**Comment by hanneslund** [comment]
I think #36827 fixed this already.

**Comment by mdubourg001** [comment]
@hanneslund I don't think this PR fixes the same issue. I understand why the export of `/404.html` is enforced, but I'm not sure the export of `/404.html.html` should be:

https://github.com/vercel/next.js/blob/e8a8220d49434021f2fb9975a6ec7081043b97b4/packages/next/export/index.ts#L426

**Comment by SukkaW** [comment]
@mdubourg001 

The issue should be solved by #36827 already. I have forked your stackblitz reproduction, and updated Next.js to the `next@canary`, and no `404.html.html` is being exported: https://stackblitz.com/edit/nextjs-oujzwx
